ci/travis/get_hdf5_if_needed.sh

```diff
@@ -12,11 +12,16 @@ else
 	echo "using cached build"
     else
         pushd /tmp
-        wget https://www.hdfgroup.org/ftp/HDF5/releases/hdf5-1.10/hdf5-$HDF5_VERSION/src/hdf5-$HDF5_VERSION.tar.gz
+        #                             Remove trailing .*, to get e.g. '1.12' ↓
+        wget "https://www.hdfgroup.org/ftp/HDF5/releases/hdf5-${HDF5_VERSION%.*}/hdf5-$HDF5_VERSION/src/hdf5-$HDF5_VERSION.tar.gz"
         tar -xzvf hdf5-$HDF5_VERSION.tar.gz
         pushd hdf5-$HDF5_VERSION
         chmod u+x autogen.sh
-        ./configure --prefix $HDF5_DIR
+        if [[ "${HDF5_VERSION%.*}" = "1.12" ]]; then
+          ./configure --prefix $HDF5_DIR --with-default-api-version=v110
```

Comment:
This is pinning back a few extra APIs, I guess we need to either document the extra flags when building with 1.12 or sort out which ones they are and manually version them back?

Reply:
I don't think you need to set the HDF5 build to default to an older API, because h5py chooses the API version it wants explicitly via H5_USE_API_x macro. You can let hdf5 build default to latest. 

That would also be a better test from CI perspective: distro users have one hdf5 installed in their system (default build, so v112 by default currently), and they want to build h5py against that installation, without affecting other consumers of hdf5. That's exactly what I did; I didn't change the hdf5 installation for h5py.

Reply:
Oh, :sheep:,  I missed that this was the configure for hdf5 not for h5py.  I'm going to try pushing a commit dropping this flag.